### PR TITLE
Create action to manage tests container

### PR DESCRIPTION
**Pull request by AndrewLister-STFC**
## Description of work

Removes dependency on docker in favour of podman and github container registry.
Podman is the open source container solution by redhat and available on ubuntu.

I also wrote a gh action that should update the container if we merge a pr that changes the Containerfile which should help with keeping the container up to date!

## Testing Instructions

1.
2.
3.

## Checklist

- [ ] The title is of a format appropriate for a line in future release notes.
- [ ] I have added the appropriate tags to the PR.
- [ ] My PR represents one logical piece of work.
- [ ] My PR fully fixes the issue linked. If new issues have been created, what are they?
- [ ] I have added the appropriate tests to cover code that has been added.
- [ ] I have updated the documentation in the relevant places to cover the changes.

### CODE REVIEW

**Comment by AndrewLister-STFC** [comment]
`COPY` seems to fail in buildah, but just on the ci… hopefully it can be removed if we target the new ralfit.

File: Container/Containerfile [hunk 1]
```diff
@@ -1,24 +1,47 @@
-# Use the basic dependencies
-FROM fitbenchmarking/fitbenchmarking-deps:latest
+# Use ubuntu as a parent image
+FROM ubuntu:24.04
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+
+COPY ralfit.patch /RALFit/
```
Comment (AndrewLister-STFC) [inline Container/Containerfile:7]:
```suggestion
```

File: Container/Containerfile [hunk 2]
```diff
@@ -41,16 +64,19 @@ RUN apt-get install -y gfortran \
     libgdal-dev
 
 RUN git clone https://github.com/ralna/RALFit
+COPY ralfit.patch /RALFit/
+RUN git apply /RALFit/ralfit.patch
```
Comment (AndrewLister-STFC) [inline Container/Containerfile:68]:
```suggestion
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -41,7 +39,10 @@ jobs:
     name: Full System Tests (Linux)
     runs-on: ubuntu-latest
     container:
-      image: 'fitbenchmarking/fitbenchmarking-extras:latest'
+      image: 'ghcr.io/fitbenchmarking/fitbenchmarking:latest'
```
Comment (AndrewLister-STFC) [inline .github/workflows/main.yml:42]:
```suggestion
      image: 'ghcr.io/fitbenchmarking/fitbenchmarking-extras:latest'
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -64,7 +65,10 @@ jobs:
     name: Full Unit Tests (Linux)
     runs-on: ubuntu-latest
     container:
-      image: 'fitbenchmarking/fitbenchmarking-extras:latest'
+      image: 'ghcr.io/fitbenchmarking/fitbenchmarking:latest'
```
Comment (AndrewLister-STFC) [inline .github/workflows/main.yml:68]:
```suggestion
      image: 'ghcr.io/fitbenchmarking/fitbenchmarking-extras:latest'
```

**Comment by AndrewLister-STFC** [comment]
@RabiyaF This is the error I get:
```
STEP 55/90: RUN cmake -G Ninja /mantid-$MANTID_VERSION -DENABLE_MANTIDQT=OFF -DENABLE_WORKBENCH=OFF -DENABLE_DOCS=OFF -DENABLE_PRECOMMIT=OFF
-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/bin/git (found version "2.43.0")
-- Found Python: /opt/venv/bin/python3.12 (found suitable version "3.12.5", minimum required is "3.6") found components: Interpreter Development NumPy Development.Module Development.Embed
-- DIST:   CODENAME:
-- Found JemallocLib: /usr/lib/x86_64-linux-gnu/libjemalloc.so
-- No build type selected, default to Release.
-- Added target ('check') for unit tests
-- Version: 6.10.0
-- Major: 6
-- Minor: 10
-- Patch: 0
-- Tweak:
-- Found CxxTest: /mantid-6.10.0/Testing/Tools/cxxtest
CMake Warning (dev) at /usr/share/cmake-3.30/Modules/FetchContent.cmake:1953 (message):
  Calling FetchContent_Populate(googletest) is deprecated, call
  FetchContent_MakeAvailable(googletest) instead.  Policy CMP0169 can be set
  to OLD to allow FetchContent_Populate(googletest) to be called directly for
  now, but the ability to call it with declared details will be removed
  completely in a future version.
Call Stack (most recent call first):
  buildconfig/CMake/GoogleTest.cmake:25 (fetchcontent_populate)
  buildconfig/CMake/CommonSetup.cmake:47 (include)
  CMakeLists.txt:96 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found Python: /opt/venv/bin/python3.12 (found version "3.12.5") found components: Interpreter
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Looking for include file stdint.h
-- Looking for include file stdint.h - found
CMake Warning (dev) at buildconfig/CMake/CommonSetup.cmake:75 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

Call Stack (most recent call first):
  CMakeLists.txt:96 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found Boost: /usr/include (found suitable version "1.83.0", minimum required is "1.65.0") found components: date_time regex serialization filesystem system
-- Found Poco: /usr/lib/x86_64-linux-gnu/libPocoFoundation.so;/usr/lib/x86_64-linux-gnu/libPocoUtil.so;/usr/lib/x86_64-linux-gnu/libPocoXML.so;/usr/lib/x86_64-linux-gnu/libPocoNet.so;/usr/lib/x86_64-linux-gnu/libPocoCrypto.so;/usr/lib/x86_64-linux-gnu/libPocoNetSSL.so (found suitable version "1.11.0", minimum required is "1.4.6")
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libcrypto.so (found version "3.0.13")
-- Found GSL: /usr/include (found version "2.7.1")
-- NeXus version 4.4.2
CMake Error at /usr/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find Nexus (missing: NEXUS_LIBRARIES) (found suitable version
  "4.4.2", minimum required is "4.3.1")
Call Stack (most recent call first):
  /usr/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  buildconfig/CMake/FindNexus.cmake:63 (find_package_handle_standard_args)
  buildconfig/CMake/CommonSetup.cmake:91 (find_package)
  CMakeLists.txt:96 (include)
```

**Comment by AndrewLister-STFC** [comment]
@RabiyaF I think I've got a bit further with this, so no need to ask at the meeting tomorrow

File: docs/source/contributors/structure.rst [hunk 5]
```diff
@@ -31,26 +31,23 @@ The specific tests run are defined in a series of Bash scripts,
 which are stored in this folder.
 
 ###################
-Docker (``Docker``)
+Container (``Container``)
 ###################
```
Comment (jess-farmer) [inline docs/source/contributors/structure.rst:35]: It's very minor but the title over- and underline here is too short which causes Sphinx to throw a warning when building the docs

**Review by jess-farmer — COMMENTED** [review]
Thanks for making these changes and adding the gh Action Andrew - it all looks good to me! Just one super minor comment, then I'm happy to merge

**Review by AndrewLister-STFC — COMMENTED** [review]
Fix heading format in docs

**Review by jess-farmer — APPROVED** [review]

**Pull request merged**